### PR TITLE
feat(upgrade): add enhancement around elemental upgrade

### PR DESCRIPTION
**Pull request by starbops**
Signed-off-by: Zespre Chang <zespre.chang@suse.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

We need more information about the execution of `elemental upgrade`.

**Solution:**

- Set up log file for `elemental upgrade`, the log file will be put under `/usr/local/upgrade_tmp/` on each node.
- Clean up downloaded large files (squashfs) before exit

**Related Issue:**

#3069

**Test plan:**

1. Prepare a v1.0.3 Harvester cluster with any number of nodes
2. Kickstart the upgrade to v1.1.0 (contains the changes in this PR)
3. After the upgrade succeeded, check the log file under `/usr/local/upgrade_tmp/elemental-upgrade-*.log`, and there shouldn't be any squashfs file

```
node-0:~ # ls -ltrh /usr/local/upgrade_tmp/
total 12K
-rw------- 1 root root   58 Nov  3 09:27 tmp.Gxer603C2T
-rw------- 1 root root   58 Nov  3 09:33 tmp.7ho2PcdZH2
-rwxr-xr-x 1 root root 1.4K Nov  3 09:34 elemental-upgrade-20221103093405.log
```

```
node-0:~ # cat /usr/local/upgrade_tmp/elemental-upgrade-20221103093405.log
INFO[2022-11-03T09:34:05Z] Starting elemental version v0.0.14
INFO[2022-11-03T09:34:05Z] Upgrading active partition
INFO[2022-11-03T09:34:05Z] Creating file system image /run/initramfs/cos-state/cOS/transition.img
INFO[2022-11-03T09:34:05Z] Running before-upgrade hook
INFO[2022-11-03T09:34:05Z] Copying COS_ACTIVE image...
INFO[2022-11-03T09:34:36Z] Finished copying COS_ACTIVE...
INFO[2022-11-03T09:34:38Z] Running after-upgrade-chroot hook
INFO[2022-11-03T09:34:38Z] Running after-upgrade hook
INFO[2022-11-03T09:34:39Z] Backing up current active image
INFO[2022-11-03T09:34:39Z] Moving /run/initramfs/cos-state/cOS/active.img to /run/initramfs/cos-state/cOS/passive.img
INFO[2022-11-03T09:34:39Z] Finished moving /run/initramfs/cos-state/cOS/active.img to /run/initramfs/cos-state/cOS/passive.img
INFO[2022-11-03T09:34:39Z] Moving /run/initramfs/cos-state/cOS/transition.img to /run/initramfs/cos-state/cOS/active.img
INFO[2022-11-03T09:34:39Z] Finished moving /run/initramfs/cos-state/cOS/transition.img to /run/initramfs/cos-state/cOS/active.img
INFO[2022-11-03T09:34:39Z] Upgrade completed
```

### CODE REVIEW

File: package/upgrade/upgrade_node.sh [hunk 1]
```diff
@@ -5,6 +5,15 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 source $SCRIPT_DIR/lib.sh
 UPGRADE_TMP_DIR=$HOST_DIR/usr/local/upgrade_tmp
 
+trap clean_up_tmp_files EXIT
+
+clean_up_tmp_files()
+{
+  echo "Clean up large tmp files..."
+  \rm -vf "$NEW_OS_SQUASHFS_IMAGE_FILE"
+  \rm -vf "$tmp_rootfs_squashfs"
```
Comment (w13915984028) [inline package/upgrade/upgrade_node.sh]: echo "Clean up large tmp files..." -> echo "Clean up tmp files..."   

should L#433 `umount $tmp_rootfs_mount` also be executed here? to umount first, then delete

and, is it better to catch `elemental upgrade` return code first, check return code, echo, and finally clean up ?

Comment (w13915984028) [reply package/upgrade/upgrade_node.sh]: another point:

should the `upgrade_node.sh` call `cat /usr/local/upgrade_tmp/elemental-upgrade-20221103093405.log` when `elemental upgrade` return code is not 0 ? thus `kubectl get log .... ` will have the error log directly

Comment (starbops) [reply package/upgrade/upgrade_node.sh]: Good catch! Updated it, please check again. About the elemental upgrade log, basically, the contents are the same (stdout vs logfile) in the case of success. But there might be useful information when it comes to unexpected errors.

**Review by w13915984028 — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]
LGTM!

**Pull request merged**